libs/ui/saveDialog.jsx

```diff
@@ -370,53 +378,116 @@ export class LoadCsvDialog extends React.Component{
     this.setState({ url: value })
   }
 
+  handleCsvNameChange (evt) {
+    const value = evt.target.value
+    this.setState({ csvName: value })
+  }
+
   localFileUpload (fileUploadRef){
-    fileUploadRef.click()
-    this.handleClose()
+    this.refs.csvFile.click()
   }
 
   urlLoad(){
-    this.props.loadCsvUrl(this.state.url)
+    // Get the end of the url, removing the extension (if it's there)
+    const formattedUrl = this.state.url.replace(/#[^#]+$/, "").replace(/\?[^\?]+$/, "").replace(/\/$/, "");
+    const label = formattedUrl.substr(formattedUrl.lastIndexOf("/") + 1).replace('.csv', '')
+    this.setState({pageName: 'nameCsv', csvName: label, type: 'url'})
+  }
+
+  loadCsv(){
+    const file = this.refs.csvFile.files[0]
+    const name = file.name
+    const label = name.replace('.csv', '')
+    this.setState({pageName: 'nameCsv', csvName: label, type: 'localFile'})
+  }
+
+  submit(){
+      if (this.state.type == 'localFile'){
+        this.submitLoadCsv()
+      } else if (this.state.type == 'url'){
+        this.submitLoadUrlCsv()
+      }
+  }
+
+  submitLoadCsv(){
+    this.props.loadCsv(this.refs.csvFile.files[0], this.state.csvName)
+    this.handleClose()
+  }
+
+  submitLoadUrlCsv(){
+    this.props.loadCsvUrl(this.state.url, this.state.csvName)
     this.handleClose()
   }
 
+
   render () {
     return (
       <Dialog open={this.state.open} onClose={this.handleClose} aria-labelledby="form-dialog-title">
+        <input type="file" id="csvFile" ref="csvFile"
+          onChange={this.loadCsv}
+          style={{display: "none"}}/>
+
         <DialogTitle id="form-dialog-title">Load CSV</DialogTitle>
+        { this.state.pageName == 'default' ?
+          <>
```

Comment:
I don't know what this is, but that's probably my limited knowledge of React.

Reply:
Empty fragment, so with React we need one element wrapping everything there. The <> shortthand maps to <React.Fragment>. Avoid creating a DOM element we don't need, while satisfying React.